pyo3-macros-backend/src/utils.rs

```diff
@@ -67,14 +68,58 @@ pub fn option_type_argument(ty: &syn::Type) -> Option<&syn::Type> {
     None
 }
 
+// TODO: Replace usage of this by [`syn::LitCStr`] when on MSRV 1.77
+#[derive(Clone)]
+pub struct LitCStr {
+    lit: CString,
+    span: Span,
+    #[allow(dead_code)]
+    ctx: Ctx,
```

Comment:
Rather than cloning the whole `Ctx` (which might grow in future), should we store just the PyO3 path here? Or take the cfg by reference?

Reply:
Taking a lifetime on `LitCStr` is a bit annoying and feels unnecessary, but just cloning out the PyO3 path sounds reasonable to me 👍